pkgs/development/python-modules/netcdf4/default.nix

```diff
@@ -26,21 +39,30 @@ buildPythonPackage rec {
     hdf5
     curl
     libjpeg
-  ];
+  ] ++ (if mpiSupport then [mpi4py mpi openssh] else []);
+
+  #patches=[./skipDapTest.patch];
+
+  makeFlags = lib.optionals mpiSupport [ "CC=mpicc" "CXX=mpicxx" ];
 
-  checkPhase = ''
-    py.test test/tst_*.py
+  checkPhase= ''
+    cd test
+    rm tst_dap.py # does not fail in the nix-shell  
+    python -m unittest tst_*.py
   '';
 
-  # Tests need fixing.
-  doCheck = false;
 
   # Variables used to configure the build process
   USE_NCCONFIG="0";
-  HDF5_DIR = lib.getDev hdf5;
-  NETCDF4_DIR=netcdf;
-  CURL_DIR=curl.dev;
-  JPEG_DIR=libjpeg.dev;
+  HDF5_DIR= lib.getDev hdf5;
+  NETCDF4_DIR="${netcdf}";
+  CURL_DIR="${curl.dev}";
+  JPEG_DIR="${libjpeg.dev}";
```

Comment:
Thanks for your time. I learned a bit about nix (I can now use inherit properly ;-) and also see the point of lib.optionals..)
I have committed all the suggestions and added the remaining outdated ones manually.
I have some remaining questions:

1. I am sorry for having created such a lot of style problems.
   Is there a linter (or at least a style guide)?
   I am using vim with vim-nix 
   For python code I use the python language server integration which is very verbose
   Is there something similar available for nix?
2. I am a bit unsure about the workflow used to deal with pull requests. 
( I normally work in a very small group at a research institute where 99.999 % of git usage consists of pull and push and even branches are an exception..., so we are not very professional users..)
 I know what rebasing means and did it twice already for this branch, but  I do not know for instance if you would prefer my (many tiny) commits to be squashed into one or if I would make things harder by it...)
So any suggestions about using git professionally are also welcome.

Thanks again.

Reply:
@mamueller 
Regarding your questions:

1. There are couple linters that could help. `nixpkgs-fmt` is one of the most promising ones. There are several other ones: https://github.com/nix-community/nixpkgs-fmt#formatters
Note that if will format whole file at the time, which may hinder the review process if you mix functional changes with style changes. A typical solution to that is to format the original file in a first commit and then introduce functional changes in subsequent separate commits. It is a bit too late to do that in this PR, but you could try this in the future.

2. Usually commits are supposed to have one specific idea for a change that is well described in the commit message. So if a PR contains several separate changes, we usually ask for them to be in separate commits. In case of this PR there is only one change  (adding mpi support to pythonPackages.netcdf4), so we might just squash everything on merge for you. If you are unsure what to do, look around at other peoples contributions, see what they do.